### PR TITLE
Feature/set components

### PR DESCRIPTION
**Pull request by MarilenaRoque**
### 2 Milestone Requirements Completed : :rocket:

- [x] Set files and folders structure, dividing, reducers, actions, containers and presentational components.
- [x]  Index renders App component, which wraps BookList and BookForm
- [x]  Booklist fetch the store to grab the books and display them on a table, where each book is represented in a row.
- [x]  Book component presents a book and receives a book as prop.
- [x] BookForm contains a text input and a select box for the category. The options should be generated by a constant array that stores the categories.
- [x] Build books reducer that handles CREATE_BOOK and REMOVE_BOOK actions
- [x]  Set CREATE_BOOK and REMOVE_BOOK actions.
- [x] Set store with an initial state.
- [x]  Pass the store to App component using Provider.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import BooksForm from '../containers/BooksForm';
+import BooksList from '../containers/BooksList';
+
+function App() {
```
Comment (christianotieno) [inline src/components/App.js]: ## Convert to ES6 syntax
- [x] We need to ensure we are using ES6 syntax all throughout the codebase. Kindly convert this `App` component function to an arrow based function.
```suggestion
const App = () => {
```

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Book({ book }) {
```
Comment (christianotieno) [inline src/components/Book.js]: ## Convert to ES6 syntax
- [x] We need to ensure we are using ES6 syntax all throughout the codebase. Kindly convert this `Book` component function to an arrow based function.
```suggestion
export default const Book = ({ book }) => {
```

File: src/containers/BooksForm.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+
+export default function BooksForm() {
```
Comment (christianotieno) [inline src/containers/BooksForm.js]: ## Convert to ES6 syntax
- [x] We need to ensure we are using ES6 syntax all throughout the codebase. Kindly convert this `BooksForm` component function to an arrow based function.
```suggestion
export default const BooksForm = () => {
```

File: src/containers/BooksList.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Book from '../components/Book';
+
+function BooksList({ bookList }) {
```
Comment (christianotieno) [inline src/containers/BooksList.js]: ## Convert to ES6 syntax
- [x] We need to ensure we are using ES6 syntax all throughout the codebase. Kindly convert this `BookList` component function declaration to an arrow based function.
```suggestion
const BooksList = ({ bookList }) => {
```

File: src/reducers/books.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    category: 'Sci-Fi',
+  },
+  {
+    id: 5,
+    title: '1922',
+    category: 'Horror',
+  },
+];
+
+function removeBookFromState(id) {
```
Comment (christianotieno) [inline src/reducers/books.js]: ## Convert to ES6 syntax
- [x] We need to ensure we are using ES6 syntax all throughout the codebase. Kindly convert this `removeBookFromState` reducer function declaration to an arrow based function.
```suggestion
const removeBookFromState = (id) => {
```

File: src/reducers/books.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  return function newState(book) {
```
Comment (christianotieno) [inline src/reducers/books.js]: ## Convert to ES6 syntax
- [ ] We need to ensure we are using ES6 syntax all throughout the codebase. Kindly convert this `newState` reducer function declaration to an arrow based function.
```suggestion
  return const newState = (book) => {
```

**Review by christianotieno — CHANGES_REQUESTED** [review]
## Changes Required ♻️ 
Hello @MarilenaRoque 👋 
The code looks neat and deliberately functioning. Good job.
Below I have highlighted some required ES6 syntax conversion, have a look 👇 
### Happy coding 💻 
#### Christian Otieno (CR)

**Comment by MarilenaRoque** [comment]
# Thank you so much for the review, @christianotieno! :rocket:

### The changes requested were made: 🦝 

- [x]  The code was refactored to apply the ES6 syntax.

**Review by whiz25 — APPROVED** [review]
## Approved ✅ 
Nice work @MarilenaRoque . Your codebase now conforms to ES6 syntax as requested. Merge and proceed to the next project.


### Martin Afani

**Pull request merged**